pycaret/containers/models/time_series.py

```diff
@@ -856,6 +856,69 @@ def _set_tune_distributions(self) -> Dict[str, List[Any]]:
             }
         return tune_distributions
 
+class CrostonContainer(TimeSeriesContainer):
+    model_type = TSModelTypes.CLASSICAL
+
+    def __init__(self, globals_dict: dict) -> None:
+        logger = get_logger()
+        np.random.seed(globals_dict["seed"])
+        self.gpu_imported = False
+
+        from sktime.forecasting.croston import Croston  # type: ignore
+
+        dummy = Croston()
+        self.active:bool = self.disable_pred_int_enforcement(
+            forecaster=dummy, enforce_pi=globals_dict["enforce_pi"]
+        )
+        if not self.active:
+            return
+
+        # self.seasonality_present = globals_dict.get("seasonality_present")
+        # TODO: Not sure if this is required.
+        sp = globals_dict.get("seasonal_period")
+        self.sp = sp if sp is not None else 1
+
+        self.strictly_positive = globals_dict.get("strictly_positive")
```

Comment:
Can `crostons` support negative values? If it can, then this may not be needed. We usually use it to disable things like multiplicative seasonality when there are negative values.

Reply:
It doesn't seem to be able to use negative values (see notebook example)

Reply:
I meant, if the data has negative values, will the model run. If it still runs with negative y values, then you do not need the `self.strictly_positive`. Anyway, I do not see you using it anywhere in the container. You can introduce some negative values in the data and see if it runs. If it does, you can remove this statement.

Reply:
I have added negative values in the example notebook, however it doesn't really handle them well / work as expected.
It doesn't raise an error, but just writes new forecasts as 0.0.

Reply:
The question that you will have to answer is what do you do when the data has negative values? Do you disable croston model or leave it enabled? If you want to leave it enabled, then there is not need to include self.strictly_positive

If you want to take some action when negative values exist (e.g. disable the model or limit the search space somehow), then you need to define this and take the appropriate action in the container.

So what do you want to do in this case?

Reply:
I think we should disable the model for time series with negative values - it isn't really relevant in the primary use case (inventory management).
if we add back in `self.strictly_positive`, then this should disable the model for data sets with negative values?

Reply:
OK, sounds good. You need to set `self.strictly_positive` first as it is currently done. Then check if it is False. If it is, then set container `active = False` and return. This is similar to what is done when checking for Prediction Interval Enforcement in the same container so you can use that as a reference.

Reply:
I have found that the 'strictly positive' test in setup.py is returning `False` when there are zeros in the data, (which disables the model, and it's main use is for intermittent forecasting!). 
So, whilst the model can't really handle negatives, it does need to handle zeros.
So using 'strictly_positive' doesn't really work in this case. 

The SKLearn algorithm just forecasts 0.0 if you feed it negative values, with no errors, which is what I suggest we replicate (I think the metrics will be uncompetitive enough that in 'compare models' it wouldn't end up near the top of the pile)

Probably easier to check out the example below.


```
from pycaret.datasets import get_data
import pandas as pd
from sktime.forecasting.croston import Croston

# Get any dataset as not really important for the demo what it is as we will transform!
y = get_data("lynx", verbose=False)
# clip the dataset and reduce the scale by the same amount to make a dataset with a lot of zeros / intermittent style
clip_number = np.quantile(y,0.84)
#clip and turn negative
y_neg= -(y.clip(lower=clip_number) - clip_number )
# init model
forecaster = Croston(smoothing=0.1)
# print training data
print(y_neg)
# fit the model
forecaster.fit(y_neg)
# create prediction
y_pred = forecaster.predict(fh=[i for i in range(5)])
print(y_pred)
```

Reply:
Yes, that is related to my original comment. If the model already takes care of this, then you can leave it enabled and remove the self.strictly_positive statement.